geospaas_harvesting/ingesters.py

```diff
@@ -46,9 +46,21 @@ class Ingester():
     QUEUE_SIZE = 500
 
     def __init__(self, max_fetcher_threads=1, max_db_threads=1):
-        self.max_fetcher_threads = max_fetcher_threads
-        self.max_db_threads = max_db_threads
+        if type(max_fetcher_threads) == int:
+            self.max_fetcher_threads = max_fetcher_threads
+        else:
+            raise TypeError
+        if type(max_db_threads) == int:
+            self.max_db_threads = max_db_threads
+        else:
+            raise TypeError
```

Comment:
You need to stop adding type checks everywhere

Reply:
These are the input values that arer coming from the user of this code. And if the wrong type comes into play accidentally (for example a typo in config file), the user must be notified in order to start the harvesting code with correct input and correct style of it. right?

Reply:
Why not, but please avoid code repetition, and use `isinstance()` instead of `type()`.
Something like that:
```python
if not (isinstance(max_fetcher_threads, int) and isinstance(max_db_threads, int)):
    raise TypeError
self.max_fetcher_threads = max_fetcher_threads
self.max_db_threads = max_db_threads
```

Reply:
Done in the next commit.